c7n/resources/vpc.py

```diff
@@ -1469,6 +1471,92 @@ class resource_type(object):
         taggable = False
 
 
+@NetworkAddress.action_registry.register('release')
+class AddressRelease(BaseAction):
+    """Action to release elastic IP address(es)
+
+    Use the force option to cause any attached elastic IPs to
+    also be released.  Otherwise, only unattached elastic IPs
+    will be released.
+
+    You can filter by AllocationId, if needed
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: release-network-addr
+                resource: network-addr
+                filters:
+                  - AllocationId: ...
+                actions:
+                  - type: release
+                    force: true|false
+    """
+
+    # Schema, permissions, method process
+    schema = type_schema('release', force={'type': 'boolean'})
+    permissions = ('ec2:ReleaseAddress','ec2:DisassociateAddress',)
+
+    @staticmethod
+    def gen_assoc_spec(network_addr):
+        key_map = {
+            'vpc': 'AllocationId',
+            'standard': 'PublicIp',
+        }
+        return {key_map[network_addr[u'Domain']]: network_addr[key_map[network_addr[u'Domain']]]}
+
+    def process(self, network_addrs):
+        def process_attached(associated_addrs, force=True):
```

Comment:
this could move to a method rather than being an enclosure, the only referenced outer scope variable is the client, which could move to a param.